api/src/main/java/org/openmrs/module/drcreports/reports/DRCTx_CurrReportManager.java

```diff
@@ -0,0 +1,535 @@
+		ProgramEnrollmentCohortDefinition pecd = new ProgramEnrollmentCohortDefinition();
+		ProgramWorkflowService ps = Context.getProgramWorkflowService();
+		Program program = ps.getProgramByUuid("64f950e6-1b07-4ac0-8e7e-f3e148f3463f"); //HIV Care program
+		pecd.setPrograms(Collections.singletonList(program));
+		
+		// ART Initiated in date range
+		SqlCohortDefinition artInitiationSqlCD = new SqlCohortDefinition();
+		String artInitiationSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCTxCurrInitiation.sql");
+		artInitiationSqlCD.setQuery(artInitiationSql);
+		artInitiationSqlCD.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		artInitiationSqlCD.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		
+		// Resume ART in date range
+		SqlCohortDefinition resumeARTSqlCD = new SqlCohortDefinition();
+		String resumeARTSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCTxCurrResumeART.sql");
+		resumeARTSqlCD.setQuery(resumeARTSql);
+		resumeARTSqlCD.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		resumeARTSqlCD.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		
+		// Transfer In and PMTCT enrollment in date range
+		SqlCohortDefinition transferInPMTCTSqlCD = new SqlCohortDefinition();
+		String transferInPMTCTSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCTxCurrTransferPMTCT.sql");
+		transferInPMTCTSqlCD.setQuery(transferInPMTCTSql);
+		transferInPMTCTSqlCD.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		transferInPMTCTSqlCD.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		
+		// Patients with drugs spaning <=3months with appt dates
+		SqlCohortDefinition multiMonthSqlCD = new SqlCohortDefinition();
+		String multiMonthSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCTxCurrMultiMonth.sql");
+		multiMonthSqlCD.setQuery(multiMonthSql);
+		multiMonthSqlCD.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		multiMonthSqlCD.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		
+		// Not dead according to death attribute 
+		BirthAndDeathCohortDefinition alive = new BirthAndDeathCohortDefinition();
+		alive.setDied(false);
+		//Not dead according to death form
+		SqlCohortDefinition liveSqlCD = new SqlCohortDefinition();
+		String liveSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCLivePatients.sql");
+		liveSqlCD.setQuery(liveSql);
+		
+		// Not stopped ART in date range
+		SqlCohortDefinition notStoppedARTSqlCD = new SqlCohortDefinition();
+		String notStoppedARTSql = getStringFromResource("org/openmrs/module/drcreports/sql/DRCTxCurrNotStopedART.sql");
+		notStoppedARTSqlCD.setQuery(notStoppedARTSql);
+		notStoppedARTSqlCD.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		notStoppedARTSqlCD.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
```

Comment:
Does this account for the case where in one quarter, a person initiates ART, discontinues ART and then recontinues it before the end date?